### ISSUE TITLE
Weekly Update

### ISSUE THREAD

**Issue by jerbeario** [jerbeario/MasterThesis#6]
Hi @maartenmarx 

I'm almost done with the deep learning pipeline (it took longer than expected...). I've implemented [a pipeline](https://github.com/jerbeario/MasterThesis/blob/main/Model/sandbox.ipynb) in PyTorch which does:

1. Data loading: Given a hazard, it loads the relevant input maps and the hazard labels into a custom torch dataset class where samples are defined as $n \times n$ patches of the inputs centered around a cell and the label is the label of the central cell. 
2. Preprocessing:  The input feature patches are respectively scaled to the range [0, 1], while the label is modified such that all occurrences of hazard (> 0) are considered as the positive class 1.
4. Data Partitioning: [A partition map](https://github.com/jerbeario/MasterThesis/blob/main/Utils/sandbox.ipynb) is generated beforehand and randomly assigns all the cells of each country to either train, valid or test sets. Binary erosion is used to remove cells near the border whose patch might leak information between sets. 
5. Model: A custom CNN architecture has been developed based on the supervisor's previous work. It consists of a several separate convolution+pooling layers for each feature patch, which get concatenated and then connected to a dense layer for classification. (spatial attention block hasn't been implemented yet). Loss function is simple BCE with ADAM optimizer.  
6 Validation: validation set is used to score the model using f1, AUC and MAE 

I still have to implement the hyperparameter optimization with weights and biases API. I also have a few questions I am writing up for our meeting next Friday. 

Ciao!
